### PR TITLE
TST: Add tests for duplicated and drop_duplicates

### PR DESCRIPTION
**Pull request by mproszewska**
- [x] refers to  #15752
- [x] tests added / passed
- [x] tests duplicated and drop_duplicates for period, categorical, datetimes, timedelta

### CODE REVIEW

File: pandas/tests/indexes/categorical/test_category.py [hunk 1]
```diff
@@ -353,16 +353,78 @@ def test_is_monotonic(self, data, non_lexsorted_data):
...
+
+    def _test_drop_duplicates(self, idx, keep, expected, index):
+        for k, e, i in zip(keep, expected, index):
+            tm.assert_numpy_array_equal(idx.duplicated(keep=k), e)
+            e = idx[~e]
+
+            result = idx.drop_duplicates(keep=k)
+            tm.assert_index_equal(result, e)
+
+            result = Series(idx).drop_duplicates(keep=k)
```
Comment (jbrockmendel) [inline pandas/tests/indexes/categorical/test_category.py]: if you're testing Series.drop_duplicates, this should go in tests.series.methods.test_drop_duplicates

File: pandas/tests/indexes/datetimes/test_ops.py [hunk 2]
```diff
@@ -259,9 +259,12 @@ def test_order_without_freq(self, index_dates, expected_dates, tz_naive_fixture)
         tm.assert_numpy_array_equal(indexer, exp, check_dtype=False)
         assert ordered.freq is None
 
-    def test_drop_duplicates_metadata(self):
+    @pytest.mark.parametrize(
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_ops.py]: would be ok with making this a fixture, call it freq_sample, and put in pandas/tests/indexes/conftest.py

File: pandas/tests/indexes/categorical/test_category.py [hunk 3]
```diff
@@ -353,16 +353,78 @@ def test_is_monotonic(self, data, non_lexsorted_data):
...
+            np.array([False, False, True]),
+            np.array([False, True, False]),
+            np.array([False, True, True]),
+        ]
+        index = [[0, 1], [0, 2], [0]]
+        self._test_drop_duplicates(idx, keep, expected, index)
+
+    def test_unique(self):
+
+        categories = [1, 2, 3]
```
Comment (jreback) [inline pandas/tests/indexes/categorical/test_category.py]: can you parameterize

File: pandas/tests/indexes/categorical/test_category.py [hunk 4]
```diff
@@ -353,16 +353,78 @@ def test_is_monotonic(self, data, non_lexsorted_data):
...
 
+        idx = CategoricalIndex([0, 1], categories=[2, 3], name="foo")
+        assert idx.is_unique is False
+        assert idx.has_duplicates is True
+
+        idx = CategoricalIndex([0, 1, 2, 3], categories=[1, 2, 3], name="foo")
+        assert idx.is_unique is True
+        assert idx.has_duplicates is False
+
+    def _test_drop_duplicates(self, idx, keep, expected, index):
```
Comment (jreback) [inline pandas/tests/indexes/categorical/test_category.py]: you can just do this in-line in the drop_duplicates test function

File: pandas/tests/indexes/categorical/test_category.py [hunk 5]
```diff
@@ -353,16 +353,78 @@ def test_is_monotonic(self, data, non_lexsorted_data):
...
     def test_drop_duplicates(self):
+        keep = ["first", "last", False]
+
+        categories = [[1, 2, 3], list("abc")]
+        expected = [
+            np.array([False, True, True]),
+            np.array([True, True, False]),
+            np.array([True, True, True]),
+        ]
+        index = [[0], [2], np.empty(shape=(0), dtype=int)]
```
Comment (jreback) [inline pandas/tests/indexes/categorical/test_category.py]: can you parameterize these

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexes/categorical/test_category.py (hunk 1 above)
Comment (mproszewska) [reply pandas/tests/indexes/categorical/test_category.py]: Previous tests included this kind of test. Series already has drop_duplicates tests, so I think I can delete it here.

**Comment by jreback** [comment]
can you merge master

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mproszewska very nice

**Pull request merged**